solution-base/mongodb/how_to_upgrade.md

```diff
@@ -0,0 +1,18 @@
+# MongoDB Sharded Upgrade Process
+
+## Overview
+
+Upgrading MongoDB sharded involves several steps to ensure a smooth transition to the new version. Below is a structured guide to follow for a successful upgrade (note that all along
+I will be taking as an example the upgrade from 5 to 6):
+
+## Upgrade steps
+
+
+1. Check for all available versions of the chart using `helm search repo bitnami/mongodb-sharded --versions` to check the configuration values for a specified version: `helm show values --version 6.6.7  bitnami/mongodb-sharded`
+2. Bump mongo version in `solution-base/deps.yaml` files.
+3. Bump mongodb sharded version in charts => `CHART_MONGO_SHARDED_VERSION` in the `solution-base/mongodb/Makefile` file (e.g. `CHART_MONGO_SHARDED_VERSION:="6.6.7"`)
+4. Upgrade charts to the version targetted => `make fetch-mongodb-sharded`
+5. Manually update the patches by applying the changes manually in the new upgraded charts (note that this can be applied only to patches that target mongo sharded) =>  `git diff -- solution-base/mongodb/charts/mongodb-sharded/values.yaml > solution-base/mongodb/patches/secret-name.patch` (this operation needs to be done for every patch).
```

Comment:
nit: the part about "mongo sharded" is not relevant anymore?

```suggestion
5. Manually update the patches by applying the changes manually in the new upgraded charts =>  `git diff -- solution-base/mongodb/charts/mongodb-sharded/values.yaml > solution-base/mongodb/patches/secret-name.patch` (this operation needs to be done for every patch).
```

Also note that this must be done only when the patch does not apply : as long as git able to apply the patch (and it works at runtime obviously), there is no need to proactively edit/re-export every patch.